src/com/palmergames/bukkit/towny/object/Nation.java

```diff
@@ -473,6 +473,7 @@ public void recheckTownDistance() throws TownyException {
 
 					double distance = Math.sqrt(Math.pow(capitalCoord.getX() - townCoord.getX(), 2) + Math.pow(capitalCoord.getZ() - townCoord.getZ(), 2));
 					if (distance > TownySettings.getNationRequiresProximity()) {
+						town.setNation(null);
```

Comment:
This ought to be nation.removeTown(town); 
so that it can be properly removed and fire the relevant events.

Reply:
Is nation.removeTown able to be called during iteration? Calling it during the iteration would more than likely throw a concurrent modification exception.

Reply:
It should be fine, might be worth testing it. We would usually be doing 
```
for (Town town : nation.getTowns()) { }
```

If you were worried about it you could fashion a list of towns to be removed after the loop is done with.

Reply:
Hmm, I'll probably just close this and leave it as is under the EarthMC fork.

Reply:
Just be sure to test it out, see that it doesn't cause any safe mode problems.

Reply:
It does not, it uses the same methods as nation.removeTown minus the event call.